### PR TITLE
Allow maxDate to be a year or a month (fixes #26)

### LINKED ISSUE #26: Allow maxDate to be a year or a month
We need to run the script many times as is in order to get historical data. We should be able to specify a month (2019-05) or a year (2019) and get all of the sheets for that time period.

Implementation-wise I would loop over the data collection starting from day 1 and incrementing by a week, until the new date is outisde the specified month or year.

### PR DESCRIPTION
**Pull request by past**
This does the dumbest thing possible for the actual creation of the spreadsheets, namely it uses a separate sheet/email for each one. But I figured it would at least be easier to review!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,42 +3,78 @@ const tranco = require('./tranco');
...
+
+        if (!inputDate.includes("-")) {
+            // An entire year is specified.
+            for (let i = 0; i < 52; i++) {
+                queryDates.push(getEOW(parsed));
+                parsed.setDate(parsed.getDate() + 7);
+            }
+        } else if (inputDate.indexOf("-") === inputDate.lastIndexOf("-")) {
+            // An entire month is specified.
+            for (let i = 0; i < 4; i++) {
```
Comment (miketaylr) [inline index.js]: (I'm guessing we don't really care about months with 5 weeks in them... or at least, we can get super technical about this in the future)

**Comment by miketaylr** [comment]
> it seems like there's limitations to the historical Tranco lists.

Looks like 2018-12-01 is probably when they started collecting data. So that's as far back as we can go.
`<input type="date" id="date" name="date" min="2018-12-01" class="form-control col-sm-10" max="2019-05-30">`

File: index.js (hunk 1 above)
Comment (miketaylr) [reply index.js]: thinking more about this, yeah, we probably don't care. we're interested in full weeks, not partial weeks. so neverrrrrmind. 🙈

**Comment by past** [comment]
I have been getting empty lists today, hence the extra error handling in the patch, but I hadn't figured out how far back they go. I've been getting empty lists in 2019, too. For 2018 we should just use the earliest available as a static list for all prior dates.

**Comment by miketaylr** [comment]
> I'll follow a file up bug to handle the error:

https://github.com/past/tsci/issues/31 filed as a follow up.

**Comment by miketaylr** [comment]
> For 2018 we should just use the earliest available as a static list for all prior dates.

Yeah, that sounds reasonable. Feel free to merge here and we can fix that up in #31.

File: index.js (hunk 1 above)
Comment (past) [reply index.js]: Right, that was my thinking, but I think you make a good point that I missed a case like March 2019. I'll fix this.

**Pull request merged**